server/services/chat/olly_chat_service.ts

```diff
@@ -81,12 +67,55 @@ export class OllyChatService implements ChatService {
     } catch (error) {
       throw error;
     } finally {
-      if (payload.sessionId) {
-        OllyChatService.abortControllers.delete(payload.sessionId);
+      if (payload.memory_id) {
+        OllyChatService.abortControllers.delete(payload.memory_id);
       }
     }
   }
 
+  public async requestLLM(
+    payload: { messages: IMessage[]; input: IInput; sessionId?: string; rootAgentId: string },
+    context: RequestHandlerContext
+  ): Promise<{
+    messages: IMessage[];
+    memoryId: string;
+  }> {
+    const { input, sessionId, rootAgentId } = payload;
+
+    const parametersPayload: {
+      question?: string;
+      verbose?: boolean;
+      memory_id?: string;
+    } = {
```

Comment:
What about define an interface `AgentRunPayload` and then reuse it on L19, L85, L106 so we don't need to define inline types repeatedly.
```typescript
interface AgentRunPayload {
    question?: string;
    verbose?: boolean;
    memory_id?: string;
    regenerate_interaction_id?: string;
  }
```

Reply:
Agree, will do that.